llama_stack/cli/stack/_build.py

```diff
@@ -395,7 +403,7 @@ def _run_stack_build_command_from_build_config(
         build_file_path,
         image_name,
         template_or_config=template_name or config_path or str(build_file_path),
-        run_config=run_config_file,
+        run_config=str(run_config_file) if run_config_file else None,
```

Comment:
note: `def build_image(..., run_config: str | None = None):`

instead, update `_generate_run_config` to `-> str` and declare `run_config_file: str | None = None`

Reply:
I’d rather keep `_generate_run_config` returning a `Path`, it’s the most accurate type for file operations and follows good practice by not mixing concerns. The conversion to str is only needed for the `build_image` call, so it makes sense to handle it there. This keeps the return type precise and avoids pushing unrelated formatting responsibility into the generator.

Reply:
fair enough!